ReactAndroid/src/main/java/com/facebook/react/views/text/ReactAbsoluteSizeSpan.java

```diff
@@ -7,13 +7,85 @@
 
 package com.facebook.react.views.text;
 
+import android.text.TextPaint;
 import android.text.style.AbsoluteSizeSpan;
 
 /*
  * Wraps {@link AbsoluteSizeSpan} as a {@link ReactSpan}.
  */
 public class ReactAbsoluteSizeSpan extends AbsoluteSizeSpan implements ReactSpan {
```

Comment:
This class is for mirroring https://developer.android.com/reference/android/text/style/AbsoluteSizeSpan.

There are some other styling spans already set which are used during layout than transformed when the EditText is hit. But otherwise, these should all be in a new span instead of reusing the one.

As of https://github.com/facebook/react-native/pull/35967 we also strip these spans when unneeded to mitigate the Samsung TextInput hang. Another reason to separate text alignment from text sizing spans.

Reply:
> The FontMetrics top, ascent, bottom, and descent do not update when changing the fontSize. The issue was solved by moving the logic to the ReactAbsoluteSizeSpan

I wonder what happens if we change font face. Like, if we are being hooked in a way we only know about the text properties local to, would we know about other font features which could change where ascent, baseline, etc are located?

From source it seems like all `AbsoluteSizeSpan` does on draw and on metric collection is to update the text size. But it seems like there should ideally be some way to resolve these at the right time. Looking through Android source and it is not clear though, since most are able to set something on the painting context without needing to read from it (e.g. [`SuggestionSpan`](https://cs.android.com/android/platform/superproject/+/master:frameworks/base/core/java/android/text/style/SuggestionSpan.java;l=51;bpv=0;bpt=1?q=SuggestionSpan&sq=&ss=android%2Fplatform%2Fsuperproject)).

I can see there are some functions for spans that do custom drawing, but otherwise I am wondering if there is a generic way to get correct font metrics.

Deriving from AbsoluteSizeSpan specifically will also opt these spans into being stripped, which we probably don't want.

Reply:
Thanks @NickGerleman. 

>I wonder what happens if we change font face. Like, if we are being hooked in a way we only know about the text properties local to, would we know about other font features which could change where ascent, baseline, etc are located?

I gave a quick read to the Paint.cpp (HWUI) implementation. I'm happy to further investigate the Android CPP implementation to improve the solution.
https://cs.android.com/android/platform/superproject/+/master:frameworks/base/libs/hwui/jni/Paint.cpp;l=636-641;drc=8587fcb26a2f943c41bdb5d94bf1a31140da20f4;bpv=0

>From source it seems like all AbsoluteSizeSpan does on draw and on metric collection is to update the text size. But it seems like there should ideally be some way to resolve these at the right time. Looking through Android source and it is not clear though, since most are able to set something on the painting context without needing to read from it (e.g. [SuggestionSpan](https://cs.android.com/android/platform/superproject/+/master:frameworks/base/core/java/android/text/style/SuggestionSpan.java;l=51;bpv=0;bpt=1?q=SuggestionSpan&sq=&ss=android%2Fplatform%2Fsuperproject)).

I moved the logic to CustomStyleSpan and I adapted the functionality. Nested Text with different fontSizes is correctly aligned to the top/bottom of the line.

Video tests included at https://github.com/facebook/react-native/pull/35949#issuecomment-1425542222

Reply:
More info available in PR https://github.com/facebook/react-native/pull/35704